src/constraint_system/ecc/mod.rs

```diff
@@ -12,13 +12,38 @@ pub struct Point {
     x: Variable,
     y: Variable,
 }
+
+impl Point {
+    /// Return the X coordinate of the point
+    pub fn x(&self) -> &Variable {
```

Comment:
Probably, `get_x()` would be better than just `x`. If you want  to use `x` why not just making `pub` the internal fields of the `Point`?

Reply:
Actually @vlopes11 is correct here.
It's not idiomatic to have getters in Rust:
> With a few exceptions, the`get_ prefix` is not used for getters in Rust code.

If you have a private field (that for API design you don't want to expose, e.g. it has a getter but not a setter), you should name the method with the same name of the field.

See: https://rust-lang.github.io/api-guidelines/naming.html